requests_auth/authentication.py

```diff
@@ -486,7 +486,7 @@ def __init__(self, tenant_id, client_id, **kwargs):
             self,
             'https://login.microsoftonline.com/{0}/oauth2/authorize'.format(tenant_id),
             client_id=client_id,
-            response_type='id_token',
+            response_type=response_type,
```

Comment:
To avoid clients issues with clients sending None or empty, and to keep a restricted number of parameters, could you please update to the following?

```suggestion
            response_type=kwargs.pop('response_type', 'id_token'),
```

Reply:
Also, this new parameter needs to be part of documentation and README :)

Reply:
I started with the pop approach but with a step back it looked a bit unpythonic (default value not explicit in signature). The doc is also a bit weird as it list parameters from the base class, correct?

Reply:
Retrieving the value from key value args makes the calls way more robust. I would agree if this was not a publicly exposed parameter but it is. Also, I am trying to keep the declarations as small as possible. Modern IDE will show you the documentation for your parameter anyway so you will have the information. From my pov there is no loss here, only gain. While when choosing the default parameter value in definition, you have the issue I mentioned earlier.

The documentation is indeed listing every parameter that are available and make sense. So that every parameter can be displayed without the need to browse the code. If you know some python equivalent to the way javadoc works by referencing super class I am all in :-)